x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_tables.tsx

```diff
@@ -140,8 +141,19 @@ export const RulesTables = React.memo<RulesTableProps>(({ selectedTab }) => {
     [setPage, setPerPage, setSortingOptions]
   );
 
-  const rulesColumns = useRulesColumns({ hasCRUDPermissions: hasPermissions });
-  const monitoringColumns = useMonitoringColumns({ hasCRUDPermissions: hasPermissions });
+  const { loading: isLoadingJobs, jobs: mlJobs, startMlJobs } = useStartMlJobs();
+  const rulesColumns = useRulesColumns({
+    hasCRUDPermissions: hasPermissions,
+    isLoadingJobs,
+    mlJobs,
+    startMlJobs,
+  });
+  const monitoringColumns = useMonitoringColumns({
+    hasCRUDPermissions: hasPermissions,
+    isLoadingJobs,
+    mlJobs,
+    startMlJobs,
+  });
```

Comment:
This is "props drilling" we started to be cautious about and avoid in the rules area. Can the `useStartMlJobs` be called closer to the usage, i.e. in `useEnabledColumn` etc?

Reply:
It was inside `useRulesColumns` and `useMonitoringColumns` before, but @vitaliidm noticed that we do multiple unnecessary calls, so we decided to move it out and reuse in both methods.

Reply:
Ah I see! We need to create a react context for the ML jobs logic, but it would be a different story for a separate PR.

Reply:
To make it looks nicer, I could add another method inside useColumns.tsx to wrap these two calls and return rulesColumns and monitoringColumns